### ISSUE TITLE
R: Shapley Feature Importances

### ISSUE THREAD

**Issue by StatMixedML** [catboost/catboost#426]
Problem: {fstr_type = "FeatureImportance" and fstr_type = "ShapValues" are identical}
catboost version: {catboost_0.9.1.1}
Operating System: { x86_64-redhat-linux-gnu (64-bit)}

Dear Catboost team,

it seems as if fstr_type = "FeatureImportance" and fstr_type = "ShapValues" are identical for the R version. 

library(catboost)
train_path <- system.file("extdata", 
                         "adult_train.1000", 
                         package="catboost")
test_path <- system.file("extdata", 
                        "adult_test.1000", 
                        package="catboost")

column_description_vector = rep('numeric', 15)
cat_features <- c(3, 5, 7, 8, 9, 10, 11, 15)
for (i in cat_features)
  column_description_vector[i] <- 'factor'

train <- read.table(train_path, 
                    head = F, 
                    sep = "\t", 
                    colClasses = column_description_vector, 
                    na.strings='NAN')
test <- read.table(test_path, 
                   head = F, 
                   sep = "\t", 
                   colClasses = column_description_vector, 
                   na.strings='NAN')
target <- c(1)
train_pool <- catboost.load_pool(data=train[,-target], 
                                 label = train[,target])
test_pool <- catboost.load_pool(data=test[,-target], 
                                label = test[,target])

fit_params <- list(iterations = 100,
                   thread_count = 10,
                   loss_function = 'Logloss',
                   ignored_features = c(4,9),
                   border_count = 32,
                   depth = 5,
                   learning_rate = 0.03,
                   l2_leaf_reg = 3.5,
                   train_dir = 'train_dir')
model <- catboost.train(pool, test_pool, fit_params)

shapl_import <- catboost.get_feature_importance(model, 
                                                    fstr_type = "ShapValues")
feature_import <- catboost.get_feature_importance(model, 
                                                  fstr_type = "FeatureImportance") 

all(shapl_import == feature_import) # TRUE

**Comment by annaveronika** [comment]
Thank you for the report, we will look into this!

**Comment by dbakshee** [comment]
Hi @StatMixedML, 

For fstr_type = "ShapValues" the get_feature_importance() function should be given the training pool. 
We'll fix handling of this case in upcoming commit.

Thanks
Dima

**Comment by annaveronika** [comment]
This is fixed in our release 0.10
